### PR TITLE
Added IMREAD_IGNORE_ORIENTATION flag

### LINKED ISSUE #6348: jpeg exif reader: option to control behaviour
### Please state the information for your system
- OpenCV version: 3.1
- Host OS: All
### In which part of the OpenCV library you got the issue?
- highgui
- imread with jpeg pictures

If the picture have exif informations for rotation, imread rotate the picture with jpeg_exif reader.
But previous opencv 3.1 didn't do that. The exif information are dependent of hardware and not standardized and not sure.
### Expected behaviour

Adding a optional parameter for jpeg reading to use or not use exifs informations.
### Actual behaviour

No choice possible and need to patch sources to add choice or disable exif reading for jpeg files.
### Code example to reproduce the issue / Steps to reproduce the issue

```
#include "opencv2/core.hpp"
#include <iostream>
using namespace std;
using namespace cv;

// previously take a picture with a smartphone in landscape and in portrait.

int main()
{
    Mat img = imread("landscape.jpg", img);
    imwrite(save_land.jpg, img);

    img = imread("portrait.jpg", img);
    imwrite(save_port.jpg, img);

    return 0;
}
```

### PR DESCRIPTION
**Pull request by ArkadiuszRaj**
Resolves #6348

### This pullrequest changes

Reorganizes the imgcodecs module. Exif reader is now independent module that can be used by any codecs. But to not forward the flags to codecs, I have moved exif rotater to the loadsave.cpp file instead. may it be that in future end user shall have access to it? (however as for now it is not in shape to make it public)

Now the new RotateImage() function is called for every kind of loaded file, not only JPEG.  Implemented in imread & imreadmulti. To prevent rotation one shall pass IMREAD_IGNORE_ORIENTATION flag while opening file.

NOTE:
I have not touched the exif reader (getExif())  - as for now it can scan JPEG files only. 
Exif decoder(parse()) is universal one hence some day in future someone could update getExif() to support TIFFs as well.

### CODE REVIEW

File: modules/imgcodecs/src/loadsave.cpp [hunk 1]
```diff
@@ -227,8 +229,11 @@ static ImageEncoder findEncoder( const String& _ext )
     return ImageEncoder();
 }
 
+
 enum { LOAD_CVMAT=0, LOAD_IMAGE=1, LOAD_MAT=2 };
 
+void RotateImage(const String& filename, Mat& img);
```
Comment (alalek) [inline modules/imgcodecs/src/loadsave.cpp]: Please use "static" for internal functions.

**Comment by StevenPuttemans** [comment]
This functionality is adding overhead if [PR7539](https://github.com/opencv/opencv/pull/7539) is also included. I suggest that both authors and reviewers come up with a consensus so that we do not get double functionality under a different name. 

BTW, if you do not understand this remark, I am talking about the rotation functionality.

**Comment by ArkadiuszRaj** [comment]
Steven -  fully agree, however that mentioned PR is to me in middle of execution, with some failing builds :)

**Comment by StevenPuttemans** [comment]
> Steven - fully agree, however that mentioned PR is to me in middle of development, with some failing builds :)

True, but then again, either @vpisarev needs to decide to close down the other PR because this one fully implements a working version of that PR or @alalek needs to decide you need to use that functionality and await the merging of that PR ;)

Did you read the discussion there about the assumed CW and CCW turning?

**Comment by ArkadiuszRaj** [comment]
Yeap, however to me we are talking about two things.

PR7539 is about adding convenient function for easier life of developer while flipping image.
This PR is about fixing some backward incompatibility when exif was introduced. 

By coincidence exif rotator indeed could use new flip method. Please be aware that exif standard defines strictly kinds of allowed rotations with optional mirroring. And PR7539 is not about mirroring :)

**Comment by StevenPuttemans** [comment]
True but if I see it correct, your RotateImage function could do the same as the mentioned flip in the other PR?

**Comment by ArkadiuszRaj** [comment]
Bad naming then. This function is about reading exif and interpreting orientation flag. 
Will change name shortly to be more descriptive and not create any confusion.

**Comment by StevenPuttemans** [comment]
Okay! Seems to me that if people would search on rotate image in the docs, they cannot get your function xD

**Comment by ArkadiuszRaj** [comment]
knock knock...

**Comment by StevenPuttemans** [comment]
There are still build errors. But I am pretty convinced it is not due to your changes. Just be patient and wait until a core dev reboots the build servers.

**Comment by alalek** [comment]
Looks good to me! :+1:

**Pull request merged**